### ISSUE TITLE
`json` reporter no longer listed 

### ISSUE THREAD

**Issue by trotzig** [brigade/haml-lint#185]
When I run `haml-lint --show-reporters`, I no longer see `json` in the list. 

Before (on version 0.21.0): 
```bash
⨠ haml-lint --show-reporters
Available reporters:
 - checkstyle
 - default
 - json
```
After installing version 0.22.0: 
```bash
⨠ haml-lint --show-reporters
Available reporters:
 - checkstyle
 - default
 - hash
 - progress
```

**Comment by trotzig** [comment]
Ping @michaelherold, I wonder if this is something that was introduced with the addition of the hash reporter?

**Comment by michaelherold** [comment]
Hmm, yeah that's definitely related. I'll need to see how that lookup works.

**Comment by michaelherold** [comment]
Urgh, it looks at `HamlLint::Reporter.descendants`. That's the issue.

Maybe a module to extract the formatting behavior makes more sense in this case.

**Comment by trotzig** [comment]
The value of the `--show-reporters` flag is questionable in my opinion. We could take the approach of removing it and just listing the values in the README instead. I doubt that there are external tools relying on the flag.

**Comment by michaelherold** [comment]
Even better would be to output it in the help for `-r, --reporter`.

**Comment by trotzig** [comment]
Fixed by 2e897e2d43.
